src/ResponseBodyHandler.cpp

```diff
@@ -0,0 +1,65 @@
+#include "ResponseBodyHandler.hpp"
+
+/**
+ * @brief Construct a new ResponseBodyHandler object
+ *
+ * @param response HTTP response.
+ * @param fileSystemPolicy File system policy. Can be mocked if needed.
+ */
+ResponseBodyHandler::ResponseBodyHandler(HTTPResponse& response, const FileSystemPolicy& fileSystemPolicy)
+	: m_response(response)
+	, m_fileSystemPolicy(fileSystemPolicy)
+{
+}
+
+/**
+ * @brief Create the response body.
+ *
+ * Depending on the HTTP Response object status, the body will be created.
+ * If the status is not OK, an error page will be created.
+ * If the status is OK, the body will be created based on the target resource.
+ * If the target resource is a directory, and autoindex is on an autoindex will be created.
+ * If the target resource is a file, the file contents will be read and set as the body.
+ * @todo FIXME: Implement other methods than GET.
+ */
+void ResponseBodyHandler::execute()
+{
+	if (m_response.status != StatusOK) {
+		handleErrorBody();
+		return ;
+	}
+	if (m_response.autoindex) {
+		AutoindexHandler autoindexHandler(m_fileSystemPolicy);
+		m_response.body = autoindexHandler.execute(m_response.targetResource);
+		if (m_response.body.empty()) {
+			m_response.status = StatusInternalServerError;
+			handleErrorBody();
+			return ;
+		}
+		m_response.status = StatusOK;
+		m_response.targetResource += "autoindex.html";
+		return ;
+	}
+	if (m_response.method == "GET") {
+		try {
+			m_response.body = m_fileSystemPolicy.getFileContents(m_response.targetResource.c_str());
+		} catch (std::exception& e) {
+			m_response.status = StatusInternalServerError;
+			handleErrorBody();
+		}
+	}
+}
+
+/**
+ * @brief Get the error page.
+ *
+ * @param status Status code.
+ * @return std::string Error page.
+ * @todo FIXME: Implement custom error pages.
+ */
+void ResponseBodyHandler::handleErrorBody()
+{
+	m_response.body = utils::getDefaultErrorPage(m_response.status);
+	// This is just to get the right extension. Could be put into its own data field of HTML struct.
```

Comment:
This class is just a stump right now, because it just returns a defaultErrorPage every time. Since I don't know how the interface for the custom error pages would look like (how are they stored), its hard to program this part.

Thats why it has a FIXME comment

Reply:
Yes, this is evident and not the cause of the implicit question:

What do you mean by the "right extension"? And a "data field of HTML structs"?

Reply:
Preface: this is just a comment for me so that I don't forget. This is not finished. It is a work in progress.

> What do you mean by the "right extension"? 

ResponseBuilder appends header Content-Type depending on located target resource, by checking the file extension. 

In case of error we don't have a targetResource - we had an error, we can't deliver. So we deliver an error page. If no custom error page is set, default error page is sent. The default error page does not exist as a file, it has no file extension. So I just set targetResource to error.html to get the right extension - ".hmtl".

> And a "data field of HTML structs"?

It says "data field of HTML struct." I could also make a std::string in HTTP Response struct. Then init it to ".html". Then overwrite if necessary.

Reply:
Discussed and cleared up.